e2e_samples/parking_sensors/infrastructure/modules/synapse.bicep

```diff
@@ -0,0 +1,179 @@
+  tags: {
+    DisplayName: 'SQL Server'
+    Environment: env
+  }
+  properties: {
+    administratorLogin: sql_server_username
+    administratorLoginPassword: sql_server_password
+  }
+  
+  resource synapse_dedicated_sql_pool 'databases@2021-02-01-preview' = {
+    name: 'syndp${env}${deployment_id}'
+    location: location
+    tags: {
+      DisplayName: 'SQL Dedicated Pool'
+      Environment: env
+    }
+    sku: {
+      name: 'DW100c'
+      tier: 'DataWarehouse'
+    }
+    properties: {
+      collation: 'SQL_Latin1_General_CP1_CI_AS'
+    }
+  }
+
+  resource firewall_rules 'firewallRules@2021-02-01-preview' = {
+    name: 'AllowAllAzureIps'
+    properties: {
+      endIpAddress: '0.0.0.0'
+      startIpAddress: '0.0.0.0'
+    }
+  }
+}
+
+resource synapseWorkspace 'Microsoft.Synapse/workspaces@2021-03-01' = {
+  name: 'syws${env}${deployment_id}'
+  tags: {
+    DisplayName: 'Synapse Workspace'
+    Environment: env
+  }
+  location: location
+  identity: {
+    type: 'SystemAssigned'
+  }
+  properties: {
+    defaultDataLakeStorage: {
+      accountUrl: 'https://${synStorage.name}.dfs.${environment().suffixes.storage}'
+      filesystem: synFileSystem.name
+    }
+    publicNetworkAccess: 'Enabled'
+    managedVirtualNetwork: 'default'
+    managedResourceGroupName: '${project}-mrg-${env}-syn'
+    sqlAdministratorLogin: 'sqladminuser'
+    sqlAdministratorLoginPassword: ''
+  }
+}
+
+resource synapse_spark_sql_pool 'Microsoft.Synapse/workspaces/bigDataPools@2021-03-01' = {
+  parent: synapseWorkspace
+  name: 'synsp${env}${deployment_id}'
+  location: location
+  tags: {
+    DisplayName: 'Spark SQL Pool'
+    Environment: env
+  }
+  properties: {
+    isComputeIsolationEnabled: false
+    nodeSizeFamily: 'MemoryOptimized'
+    nodeSize: 'Small'
+    autoScale: {
+      enabled: true
+      minNodeCount: 3
+      maxNodeCount: 10
+    }
+    dynamicExecutorAllocation: {
+      enabled: false
+    }
+    autoPause: {
+      enabled: true
+      delayInMinutes: 15
+    }
+    sparkVersion: '2.4'
+    sessionLevelPackagesEnabled: true
+    customLibraries: []
+    defaultSparkLogFolder: 'logs/'
+    sparkEventsFolder: 'events/'
+  }
+}
+
+resource synapseWorkspaceFirewallRule1 'Microsoft.Synapse/workspaces/firewallrules@2021-03-01' = {
+  parent: synapseWorkspace
+  name: 'allowAll'
+  properties: {
+    startIpAddress: '0.0.0.0'
+    endIpAddress: '255.255.255.255'
+  }
+}
+
+resource roleAssignment1 'Microsoft.Authorization/roleAssignments@2020-08-01-preview' = {
+  name: guid(resourceGroup().id, resourceId('Microsoft.Storage/storageAccounts', synStorage.name))
+  properties: {
+    principalId: synapseWorkspace.identity.principalId
+    roleDefinitionId: subscriptionResourceId('Microsoft.Authorization/roleDefinitions', 'ba92f5b4-2d11-453d-a403-e96b0029c9fe')
+    principalType: 'ServicePrincipal'
+  }
+  scope: synStorage
+}
+
+resource roleAssignment2 'Microsoft.Authorization/roleAssignments@2020-08-01-preview' = {
```

Comment:
Just copy-pasting my previous comment in the older PR #386 :)

Could we rename to something more meaningful like `roleAssignemntSynStorage` ?

Also would be good to set the roleDefinitionId to variables names that describe the role so its much more readable than just a GUID.  
Something like this: https://github.com/Azure-Samples/modern-data-warehouse-dataops/blob/226e0443a009a145e8583de574b973954d82d8c7/e2e_samples/parking_sensors/infrastructure/modules/storage.bicep#L13

Reply:
Both corrected. Tks Lace.
I've also ran into an issue with your main storage account deployment. I've added the "principalType" property  and that solved the problem.